### ISSUE TITLE
Showing a notice before the benchmark

### ISSUE THREAD

**Issue by tillkruss** [phpbench/phpbench#1003]
Is there any mechanism to show a colonized notice/warning once before executing any test?

**Comment by dantleech** [comment]
Not sure what you mean? What's the use case?

**Comment by dantleech** [comment]
Not really, you can add an "environment" sampler to add some env data which you could use in a Report:

https://github.com/phpbench/phpbench/blob/master/lib/Environment/Provider/Php.php

**Comment by dantleech** [comment]
and you _could_ add a custom progress logger that could show a warning, but maybe overkill (the env data is in `$suite`):

https://github.com/phpbench/phpbench/blob/236a04073e0dff1e043e3e8c564dbdc4562a046d/lib/Progress/Logger/PhpBenchLogger.php#L65-L70

**Comment by dantleech** [comment]
and you _could_ add a custom progress logger that could show a warning, but maybe overkill (the env data is in `$suite`):

https://github.com/phpbench/phpbench/blob/236a04073e0dff1e043e3e8c564dbdc4562a046d/lib/Progress/Logger/PhpBenchLogger.php#L65-L70

**Comment by dantleech** [comment]
and you _could_ add a custom progress logger that could show a warning, but maybe overkill (the env data is in `$suite`):

https://github.com/phpbench/phpbench/blob/236a04073e0dff1e043e3e8c564dbdc4562a046d/lib/Progress/Logger/PhpBenchLogger.php#L65-L70

**Comment by tillkruss** [comment]
That is a big overkill. Any chance you want to add New Relic to the Xdebug/Opcache list? It has a significant performant impact when running benchmarks.

**Comment by dantleech** [comment]
Maybe , although needs more thought. What is there is a bit hacky, and wouldn't want to show "New Relic" on each run as we with XDebug/OPCache.
